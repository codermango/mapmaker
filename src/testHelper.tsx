import { render } from "@testing-library/react";
import { ReactElement } from "react";
import { DndProvider } from "react-dnd";
import { HTML5Backend } from "react-dnd-html5-backend";
import { Provider } from "react-redux";
import { store } from "./app/store";

export const renderWithRedux = (component: ReactElement) => {
  return render(<Provider store={store}>{component}</Provider>);
};

export const renderWithDnd = (component: ReactElement) => {
  return render(<DndProvider backend={HTML5Backend}>{component}</DndProvider>);
};

export const renderWithReduxAndDnd = (component: ReactElement) => {
  return render(
    <Provider store={store}>
      <DndProvider backend={HTML5Backend}>{component}</DndProvider>
    </Provider>
  );
};
