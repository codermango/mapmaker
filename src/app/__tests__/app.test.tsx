import { renderWithReduxAndDnd } from "../../testHelper";
import App from "../app";

test("renders correctly", () => {
  expect(renderWithReduxAndDnd(<App />).asFragment()).toMatchSnapshot();
});
