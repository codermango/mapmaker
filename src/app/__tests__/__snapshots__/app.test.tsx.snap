// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<DocumentFragment>
  <div
    class="app"
  >
    <div
      class="appBar"
    >
      In order for the tool to work properly, please make sure your broswer zoom is 100%
    </div>
    <div
      class="content"
    >
      <div
        class="toolSidebar"
      >
        <div
          class="sidebar"
        >
          <div
            class="categories"
          >
            <div
              class="categoryItem active"
            >
              Flag
            </div>
            <div
              class="categoryItem"
            >
              Vehicle
            </div>
            <div
              class="categoryItem"
            >
              Aircraft
            </div>
            <div
              class="categoryItem"
            >
              Ship
            </div>
          </div>
          <div
            class="categoryItem deleteIcon"
          >
            <img
              alt="dustbin"
              class="deleteImg"
              src="delete.png"
            />
          </div>
        </div>
        <div
          class="assetsBoard"
        >
          <div
            class="element"
          >
            <img
              alt="flag1"
              draggable="true"
              src="flag1.png"
            />
          </div>
          <div
            class="element"
          >
            <img
              alt="flag2"
              draggable="true"
              src="flag2.png"
            />
          </div>
          <div
            class="element"
          >
            <img
              alt="flag3"
              draggable="true"
              src="flag3.png"
            />
          </div>
          <div
            class="element"
          >
            <img
              alt="flag4"
              draggable="true"
              src="flag4.png"
            />
          </div>
        </div>
      </div>
      <div
        class="droppableMap"
      >
        <div
          class="mapWrapper"
        >
          <img
            alt="map"
            class="map"
            src="map.png"
          />
        </div>
      </div>
    </div>
    <div
      class="output"
    >
      <div
        class="action"
      >
        <button
          class="button"
        >
          Display
        </button>
        <a
          class="button"
          download="level.json"
          href="data:application/octet-stream,[]"
        >
          Save level
        </a>
        <button
          class="button"
        >
          Upload level
          <input
            class="uploadBtn"
            id="file-selector"
            type="file"
          />
        </button>
      </div>
      <pre
        class="outputJson"
      />
    </div>
  </div>
</DocumentFragment>
`;
